### PR TITLE
Justin/networking: Same Networking PR + BOOM touch-ups

### PR DESCRIPTION
**Pull request by thisjustin123**
**NOTE:** (The bulk of this PR is the same as the previous, save for some new touch-ups; for some reason, the previous PR was deleted without being merged correctly.)

# New Changes:

- Added a "No classes today" section to the home page when there are no classes.
- Fixed some weird dark mode interactions.
- Updated designs for favorite buttons.
- HomeCard icons (bowling, swimming, gymnasium) now display correct information for each gym.

### Description:
This PR allows the app to connect with our GraphQL backend to display actual, relevant gym information. I used the _Apollo_ library to connect with GraphQL, which has made the integration process relatively simple.

## Changes:

- Various tweaks to the Home Card to be able to display all relevant information.
- Implementation of Apollo to make GraphQL queries.
- Creation of `UpliftApiRepository.kt` to allow for various parts of the app to interact with backend data.
- Integration of backend data into the Home Page, Gym Detail Page, and Class Detail Page.

### Demo:
https://user-images.githubusercontent.com/47724806/230204036-4b350e2f-8c5e-4e24-8951-d57e93de901a.mp4

### Next Steps:
- Still must implement loading screens for when the Api repository is still in the _Loading_ state.
- Once backend finishes their schema updates, update Apollo and adapters to match.

### CODE REVIEW

**Comment by Corfish123** [comment]
Not sure what happened to the favorite button but it looks like this on my phone. Didn't look like this in previous versions
image

File: app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt [hunk 1]
```diff
@@ -103,12 +99,30 @@ fun HomeCard(gym: UpliftGym, onClick: () -> Unit) {
...
+                                    painter = painterResource(id = R.drawable.ic_bowling_pins),
+                                    contentDescription = "Bowling Pins",
+                                    colorFilter = ColorFilter.tint(color = GRAY04)
+                                )
+                            }
+                            if (gym.swimmingInfo != null) {
+                                Spacer(Modifier.width(4.dp))
+                                Image(
+                                    painter = painterResource(id = R.drawable.ic_swimming_pool),
+                                    contentDescription = "Bowling Pins",
```
Comment (Corfish123) [inline app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt]: change the content description to what it actually is

File: app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt [hunk 2]
```diff
@@ -103,12 +99,30 @@ fun HomeCard(gym: UpliftGym, onClick: () -> Unit) {
...
+                                    colorFilter = ColorFilter.tint(color = GRAY04)
+                                )
+                            }
+                            if (gym.gymnasiumInfo != null) {
+                                Spacer(Modifier.width(4.dp))
+                                Image(
+                                    painter = painterResource(id = R.drawable.ic_basketball_hoop),
+                                    contentDescription = "Bowling Pins",
```
Comment (Corfish123) [inline app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt]: same here

**Review by Corfish123 — COMMENTED** [review]
Left some comments but overall great work!!

File: app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt (hunk 2 above)
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt]: Resolved in [277477d](https://github.com/cuappdev/uplift-android/pull/8/commits/277477d56e27c3145bbb6441590322fc88fc5741).

File: app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt (hunk 1 above)
Comment (thisjustin123) [reply app/src/main/java/com/cornellappdev/uplift/ui/components/home/HomeCard.kt]: Resolved in [277477d](https://github.com/cuappdev/uplift-android/pull/8/commits/277477d56e27c3145bbb6441590322fc88fc5741).

**Comment by Corfish123** [comment]
Great work!!

**Review by Corfish123 — APPROVED** [review]
Great work!!!

**Pull request merged**